pydantic/json.py

```diff
@@ -53,19 +53,26 @@ def pydantic_encoder(obj: Any) -> Any:
     elif is_dataclass(obj):
         return asdict(obj)
 
-    try:
-        encoder = ENCODERS_BY_TYPE[obj.__class__]
-    except KeyError:
-        raise TypeError(f"Object of type '{obj.__class__.__name__}' is not JSON serializable")
-    else:
+    # Check the class type and its superclasses for a matching encoder
```

Comment:
I think with these changes `Path` and `Enum` above can be moved into `ENCODERS_BY_TYPE`

Reply:
Done in https://github.com/samuelcolvin/pydantic/pull/1291/commits/3370971b79abe15ce09261b49d07875cac15e401